EventFiltering/PWGHF/HFFilterHelpers.h

```diff
@@ -676,38 +734,48 @@ inline bool HfFilterHelper::isSelectedProton4Femto(const T1& track, const T2& tr
     return false; // use only global tracks
   }
 
-  float NSigmaTPC = track.tpcNSigmaPr();
-  float NSigmaTOF = track.tofNSigmaPr();
-  if (!forceTof && !track.hasTOF()) {
-    NSigmaTOF = 0.; // always accepted
+  // PID evaluation
+  float NSigmaTPC = (trackSpecies == kProtonForFemto) ? track.tpcNSigmaPr() : track.tpcNSigmaDe();
+  float NSigmaTOF = (trackSpecies == kProtonForFemto) ? track.tofNSigmaPr() : track.tofNSigmaDe();
+
+  // Handle TOF usage based on the pT threshold and requirements
+  if (pt <= ptThresholdPidStrategy) {
+    NSigmaTOF = 0.; // Use only TPC below the threshold
+  } else if (!forceTof && !track.hasTOF()) {
+    NSigmaTOF = 0.; // If TOF is not required and unavailable, bypass it
   }
 
+  // Apply TPC PID post-calibration, only available for proton, dummy for deuteron
   if (mTpcPidCalibrationOption == 1) {
-    NSigmaTPC = getTPCPostCalib(track, kPr);
+    NSigmaTPC = getTPCPostCalib(track, trackSpecies == kProtonForFemto ? kPr : kDe);
   } else if (mTpcPidCalibrationOption == 2) {
     if (track.sign() > 0) {
-      NSigmaTPC = getTPCSplineCalib(track, kPr);
+      NSigmaTPC = getTPCSplineCalib(track, trackSpecies == kProtonForFemto ? kPr : kDe);
     } else {
-      NSigmaTPC = getTPCSplineCalib(track, kAntiPr);
+      NSigmaTPC = getTPCSplineCalib(track, trackSpecies == kProtonForFemto ? kAntiPr : kAntiDe);
     }
   }
 
   float NSigma = std::sqrt(NSigmaTPC * NSigmaTPC + NSigmaTOF * NSigmaTOF);
 
-  if (trackPar.getPt() <= mPtThresholdPidStrategyForFemto) {
-    if (NSigma > mNSigmaPrCutsForFemto[2]) {
+  // Apply different PID strategy in different pt range
+  if (pt <= ptThresholdPidStrategy) {
+    // Use only TPC below the threshold
```

Comment:
again this I would do only in the deuteron case, not to change the behaviour we had for the proton